File: totalnumbookpages.py
total_shelves = int(input())
total_queries = int(input())

total_num_books = [0] * total_shelves
total_num_pages = [[0] * 1100 for _ in range(total_shelves)]

for _ in range(total_queries):
    query = input().split()
    query_type = int(query[0])

    if query_type == 1:
        x = int(query[1])
        y = int(query[2])

        total_num_pages[x][total_num_books[x]] = y
        total_num_books[x] += 1
    elif query_type == 2:
        x = int(query[1])
        y = int(query[2])

        print(total_num_pages[x][y])
    elif query_type == 3:
        x = int(query[1])

        print(total_num_books[x])

File: totalnumbookpages.java
import java.util.Scanner;

public class totalnumbookpages {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int totalShelves = scanner.nextInt();
        int totalQueries = scanner.nextInt();
        
        int[] totalNumBooks = new int[totalShelves];
        int[][] totalNumPages = new int[totalShelves][1100];
        
        for (int i = 0; i < totalShelves; i++) {
            for (int j = 0; j < 1100; j++) {
                totalNumPages[i][j] = 0;
            }
        }
        
        for (int q = 0; q < totalQueries; q++) {
            int queryType = scanner.nextInt();
            
            if (queryType == 1) {
                int x = scanner.nextInt();
                int y = scanner.nextInt();
                
                totalNumPages[x][totalNumBooks[x]] = y;
                totalNumBooks[x]++;
            } else if (queryType == 2) {
                int x = scanner.nextInt();
                int y = scanner.nextInt();
                
                System.out.println(totalNumPages[x][y]);
            } else if (queryType == 3) {
                int x = scanner.nextInt();
                
                System.out.println(totalNumBooks[x]);
            }
        }
        
        scanner.close();
    }
}
